packages/pcd-collection/test/PCDCollection.spec.ts

```diff
@@ -271,6 +271,140 @@ describe("PCDCollection", async function () {
     );
     expect(await deserializedCollection.getHash()).to.eq(firstHash);
   });
+
+  it("should merge two PCD collections", async function () {
+    const pcdList = await Promise.all([
+      newPCD(),
+      newPCD(),
+      newPCD(),
+      newPCD(),
+      newPCD()
+    ]);
+
+    const firstCollection = new PCDCollection(packages);
+    const secondCollection = new PCDCollection(packages);
+
+    firstCollection.add(pcdList[0]);
+    firstCollection.add(pcdList[1]);
+    firstCollection.add(pcdList[2]);
+    secondCollection.add(pcdList[3]);
+    secondCollection.add(pcdList[4]);
+
+    firstCollection.merge(secondCollection);
+
+    expect(firstCollection.getAll()).to.deep.eq(pcdList);
+  });
+
+  it("should upsert PCDs with the same ID by default", async function () {
+    const pcdList = await Promise.all([
+      newPCD(),
+      newPCD(),
+      newPCD(),
+      newPCD(),
+      newPCD()
+    ]);
+
+    const firstCollection = new PCDCollection(packages);
+    const secondCollection = new PCDCollection(packages);
+
+    firstCollection.add(pcdList[0]);
+    firstCollection.add(pcdList[1]);
+    firstCollection.add(pcdList[2]);
+    firstCollection.add(pcdList[3]);
+    firstCollection.add(pcdList[4]);
+    secondCollection.add(pcdList[3]);
```

Comment:
I don't think you can really tell whether an upsert is happening since you're replacing a PCD with itself.  I.e. would this fail if upsert==false?

Based on the Map docs, upsert will preserve the original order, so putting secondCollection in a different order won't make a detectable difference.  I think you need to make a PCD with the same ID and different data to be sure of that.  Looks like newPCD has an `id` argument to allow that.

Reply:
Yes, it would fail if upsert==false, because the IDs conflict. `firstCollection` is a `PCDCollection` not a map, and `addAll()` (called by `add`) throws an error if you try to add a PCD with the same ID as a PCD already in the collection.

Reply:
I didn't realize the upsert==false behavior was to throw rather than ignore.  Thanks for clarifying.
Seems like that should be confirmed in a test, but that doesn't need to be this one, and hopefully already exists.